macros/src/lib.rs

```diff
@@ -790,6 +823,44 @@ fn extract_cfgs(attrs: Vec<Attribute>) -> (Vec<Attribute>, Vec<Attribute>) {
     (cfgs, not_cfgs)
 }
 
+fn check_attr_whitelist(attrs: &[Attribute]) -> Result<(), TokenStream> {
+    let whitelist = &["doc", "link_section", "cfg", "allow", "warn", "deny", "forbid", "cold"];
+    let cortex_m_rt_blacklist = &["entry", "exception", "interrupt", "pre_init"];
+
+    'o: for attr in attrs {
+        for val in whitelist {
+            if eq(&attr, &val) {
+                continue 'o;
+            }
+        }
+
+        for val in cortex_m_rt_blacklist {
+            if eq(&attr, &val) {
+                return Err(
+                    parse::Error::new(
+                        attr.span(),
+                        "cortex-m-rt does not support multiple attributes on the same function",
```

Comment:
Hmm, this wording suggests that *no* other attributes are ever allowed, but the whitelisted ones are. What about this wording instead:

```suggestion
                        "multiple cortex-m-rt attributes are not supported on the same function",
```